### PR TITLE
RELATED: BB-2494 - Add support for experimental workspace

### PR DESCRIPTION
**Pull request by lupko**
-  Experimental workspace lives on stg3 where new api features land first
-  We needs a way to interact with that workspace and make recordings from it so that we can test easily during dev
-  this adds a new package `tools/experimental-workspace`
-  a new workspace type is available for sdk-ui-tests scenarios: 'experimental-workspace'
-  the test infra was extended to allow capturing definitions into the experimental-workspace

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/479/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2070/ : FAILURE in 8m 59s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: tools/experimental-workspace/src/recordings/metadata/displayForms/label.owner.region/elements.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+    {
+        "title": "East Coast",
+        "uri": "/gdc/md/l32xdyl4bjuzgf9kkqr2avl55gtuyjwf/obj/1086/elements?id=460488"
```
Comment (no23reason) [inline tools/experimental-workspace/src/recordings/metadata/displayForms/label.owner.region/elements.json:4]: This uri is from a different project than specified in refresh-ldm.sh and refresh-recordings.sh

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/279/ : FAILURE in 2m 39s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/535/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2084/ : FAILURE in 11m 19s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/494/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2086/ : SUCCESS in 9m 45s

File: tools/experimental-workspace/package.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    "dependencies": {
+        "@gooddata/sdk-backend-spi": "^8.0.0-beta.66",
+        "@gooddata/sdk-model": "^8.0.0-beta.66",
+        "lodash": "^4.17.19",
+        "tslib": "^2.0.0"
+    },
+    "devDependencies": {
+        "@gooddata/catalog-export": "^8.0.0-beta.66",
+        "@gooddata/eslint-config": "^2.0.0",
+        "@gooddata/mock-handling": "8.0.0-beta.66",
```
Comment (no23reason) [inline tools/experimental-workspace/package.json:44]: Shall we use caret here?

Comment (lupko) [reply tools/experimental-workspace/package.json]: good catch. for some reason - typo i suspect - mock-handling was used without caret on multiple packages: all workspaces and ui-tests. fixed

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/280/ : SUCCESS in 3m 39s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/536/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2093/ : SUCCESS in 11m 05s

**Pull request merged**